### DISCUSSION TITLE
Valuing unpaid trips to prevent transfer routes from being declared unprofitable

### DISCUSSION THREAD

**Discussion by DooMy-666** [voxeltycoon/issues#335] [General]
The idea is not to be paid for factory -> warehouse trips but only to value it on the vehicle (a bit like OpenTTD for transfers)

**Comment by JezzyBelle** [comment]
I agree that it is disturbing to see some trains that supply everything to my supply chain still get complaints over lack of income.

**Comment by IDragonfire** [comment]
What is the advantage in comparison to just turn off  "unprofitable" notifcation (allready implemented)?

**Reply by DooMy-666** [reply]
This would allow true non-profitable vehicles to be seen in the list of vehicles. It would also help to understand what that trip is worth, if it makes sense to improve it with a more efficient vehicle or mode of transportation.

To date, we end up with trucks that earn far too much for their actual trip, which distorts the true profitability of a multi-transfer route.

We could also imagine that during a delivery, the payment is distributed in proportion to the journey made between the different trucks and trains that have contributed to the transport, but this seems to me to be more complex to put in place, knowing that there are transformations of intermediate products that take place in the meantime and that the factories contribute to this valuation.

**Comment by katepolak** [comment]
This is sadly basically impossible to do with how the entire game works, with materials being converted to different ones by the player instead of "NPC" factories like in other games and with the prices of items determined by the location they are delivered to.

The only way to make this work that I can think of is track all vehicles that held the item during its lifetime and split the profits between them once delivered. But this would introduce quite a bit of complexity to both the vehicle and, even more importantly, the factory system, because it would have to take into account cases like factories consuming different number of materials then they produce and so on. Not to mention the additional CPU and RAM usage from having to track this for every single item in the game, which you will have 1000s, if not more, at a time.

**Reply by JezzyBelle** [reply]
I can understand that. But maybe there can be a compromise. My primary issue when managing my fleet is that I want to be able quickly monitor those vehicles that are not profitable. When I list all those with 0 profits, those supply routes always show up adding to the clutter. Currently, I label the route with SUP for supply as a way to help sort them. Two problems with that for me: 
~By adding the label "SUP" I typically lose 4 characters that I could use to further describe that route.
~They still show up on the list as having no profit, only a loss. They still make me think something is wrong.

If I could designate a vehicle as "supply chain" or something similar and now have the 0 in the profit column just show "Supply" it would no longet show up on my profit view. In other words.... I don't wnat to see anything on my list that is not going to show a profit by design when I am analyzing things that should show a profit. The method I use still lists them.

**Comment by pippin88** [comment]
One solution:
Have warehouse or factory "buy" the goods from train at ~half market price (or player determined price).
Then warehouse sells goods to truck.
Truck takes to business and sells to business.

Train makes Warehouse buy price minus supply cost (if any)
Truck makes business buy price minus warehouse price

**Reply by tycode** [reply]
I'd like to specifically +1 this using a player determined price.

**Reply by DooMy-666** [reply]
Yes I think it could be the better and easier solution, without need of revamp the whole system.
